src/payment/strategies/nonce-pass-through/nonce-pass-through-payment-strategy.spec.ts

```diff
@@ -0,0 +1,115 @@
+import { getCheckoutStoreState } from '../../../checkout/checkouts.mock';
+import { InvalidArgumentError, MissingDataError } from '../../../common/error/errors';
+import { OrderActionCreator, OrderActionType, OrderRequestBody, OrderRequestSender, SubmitOrderAction } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentActionType, SubmitPaymentAction } from '../../payment-actions';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import PaymentRequestSender from '../../payment-request-sender';
+import PaymentRequestTransformer from '../../payment-request-transformer';
+
+import NoncePassThroughPaymentStrategy from './nonce-pass-through-payment-strategy';
+
+describe('NoncePassThroughPaymentStrategy', () => {
+    let orderActionCreator: OrderActionCreator;
+    let paymentRequestTransformer: PaymentRequestTransformer;
+    let paymentRequestSender: PaymentRequestSender;
+    let paymentActionCreator: PaymentActionCreator;
+    let options: PaymentRequestOptions;
+    let payload: OrderRequestBody;
+    let store: CheckoutStore;
+    let strategy: NoncePassThroughPaymentStrategy;
+    let submitOrderAction: Observable<SubmitOrderAction>;
+    let submitPaymentAction: Observable<SubmitPaymentAction>;
+
+    beforeEach(() => {
+        store = createCheckoutStore(getCheckoutStoreState());
+        orderActionCreator = new OrderActionCreator(
+            new OrderRequestSender(createRequestSender()),
+            new CheckoutValidator(new CheckoutRequestSender(createRequestSender()))
+        );
+        paymentRequestTransformer = new PaymentRequestTransformer();
+        paymentRequestSender = new PaymentRequestSender(createPaymentClient());
+        paymentActionCreator = new PaymentActionCreator(
+            paymentRequestSender,
+            orderActionCreator,
+            paymentRequestTransformer
+        );
+        submitOrderAction = of(createAction(OrderActionType.SubmitOrderRequested));
+        submitPaymentAction = of(createAction(PaymentActionType.SubmitPaymentRequested));
+
+        payload = {
+            payment: {
+                methodId: 'foobar',
+                paymentData: {
+                    nonce: 'fooNonce',
+                },
+            },
+        };
+        options = { methodId: 'foobar' };
+
+        jest.spyOn(store, 'dispatch');
+
+        jest.spyOn(orderActionCreator, 'submitOrder')
+            .mockReturnValue(submitOrderAction);
+
+        jest.spyOn(paymentActionCreator, 'submitPayment')
+            .mockReturnValue(submitPaymentAction);
+
+        strategy = new NoncePassThroughPaymentStrategy(store, orderActionCreator, paymentActionCreator);
+    });
+
+    it('returns checkout state when initializing the strategy', async () => {
+        return expect(strategy.initialize()).resolves.toEqual(store.getState());
```

Comment:
I just wondering why are you returning a value in a unit test

Reply:
oh yeah, remove the `async` or `await` the assertion